### PR TITLE
chore(api): refactor user endpoints

### PR DESCRIPTION
**Pull request by nmbarel**
Closes # 

## 📑 Description

refactor users endpoint to sit on /users instead of /settings, in order to fit with all conventions on other configs of keep (providers, workflows, etc)

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

### CODE REVIEW

File: keep/api/api.py [hunk 1]
```diff
@@ -159,25 +160,25 @@ def get_app(
...
-            if not user:
-                return JSONResponse(
+            if not user:                return JSONResponse(
                     status_code=401,
                     content={"message": "Invalid username or password"},
                 )
             # generate a JWT secret
             jwt_secret = os.environ.get("KEEP_JWT_SECRET")
             if not jwt_secret:
+                logger.info("invalid jwt token")
```
Comment (talboren) [inline keep/api/api.py:181]: probably better:
```suggestion
                logger.info("missing KEEP_JWT_SECRET environment variable")
```

File: keep/api/api.py [hunk 2]
```diff
@@ -159,25 +160,25 @@ def get_app(
...
-
+    if AUTH_TYPE == "SINGLE_TENANT":
         @app.post("/signin")
         def signin(body: dict):
             # validate the user/password
             user = get_user(body.get("username"), body.get("password"))
 
-            if not user:
-                return JSONResponse(
+            if not user:                return JSONResponse(
```
Comment (talboren) [inline keep/api/api.py:174]: This looks like a bug

File: keep/api/api.py [hunk 3]
```diff
@@ -159,25 +160,25 @@ def get_app(
...
     app.include_router(preset.router, prefix="/preset", tags=["preset"])
     app.include_router(groups.router, prefix="/groups", tags=["groups"])
+    app.include_router(users.router, prefix="/users", tags=["users"])
 
     # if its single tenant with authentication, add signin endpoint
     logger.info(f"Starting Keep with authentication type: {AUTH_TYPE}")
     # If we run Keep with SINGLE_TENANT auth type, we want to add the signin endpoint
-    if AUTH_TYPE == AuthenticationType.SINGLE_TENANT.value:
-
+    if AUTH_TYPE == "SINGLE_TENANT":
```
Comment (talboren) [inline keep/api/api.py:168]: Why not use the enum?

**Review by talboren — CHANGES_REQUESTED** [review]
Left 3 comments and also tests seems to be broken (probably due to the change in endpoints)

File: keep/api/api.py (hunk 3 above)
Comment (nmbarel) [reply keep/api/api.py:168]: was debugging the missing jwt and tried to see if it was due to enum failure... (before i remembered to add the env)
forgot to restore...

File: keep/api/api.py (hunk 2 above)
Comment (nmbarel) [reply keep/api/api.py:174]: yep

**Review by talboren — APPROVED** [review]
LGTM, @shahargl?

**Pull request merged**